File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcFeatureElement/IfcFeatureElementSubtraction/IfcOpeningElement/IfcOpeningElement.vb
Public Class IfcOpeningElement Inherits IfcFeatureElementSubtraction
    Public PredefinedType As IfcOpeningElementTypeEnum

    ' === INVERSE CLAUSES ===
    ' HasFillings : SET [0:?] OF IfcRelFillsElement FOR RelatingOpeningElement

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcOpeningElementTypeEnum.USERDEFINED) OR ((PredefinedType = IfcOpeningElementTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
End Class

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcFeatureElement/IfcFeatureElementSubtraction/IfcOpeningElement/IfcOpeningElement.cs
public class IfcOpeningElement : IfcFeatureElementSubtraction
{
    public IfcOpeningElementTypeEnum PredefinedType { get; set; }

    // === INVERSE CLAUSES ===
    // HasFillings : SET [0:?] OF IfcRelFillsElement FOR RelatingOpeningElement

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcOpeningElementTypeEnum.USERDEFINED) OR ((PredefinedType = IfcOpeningElementTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
}
